twindb_backup/destination/s3.py

```diff
@@ -207,39 +242,35 @@ def get_stream(self, path):
         :return:
         """
         object_key = urlparse(path).path.lstrip('/')
-        download_proc = None
 
-        def download_object(s3_client, bucket_name, key, read_fd, write_fd):
+        def _download_object(s3_client, bucket_name, key, read_fd, write_fd):
             # The read end of the pipe must be closed in the child process
             # before we start writing to it.
             os.close(read_fd)
 
-            with os.fdopen(write_fd, 'wb') as write_pipe:
-                s3_client.download_fileobj(bucket_name, key, write_pipe)
+            with os.fdopen(write_fd, 'wb') as w_pipe:
+                s3_client.download_fileobj(bucket_name, key, w_pipe)
 
-        try:
```

Comment:
You should still keep the `try` block here and remove the `except` block. So that you are sure to call `download_proc.start` in the `finally` block at all costs.